tests/test_config_parsing/test_parser_error_collection.py

```diff
@@ -0,0 +1,221 @@
+import os
+import stat
+from dataclasses import dataclass
+from typing import Callable, Dict, List, Optional, TextIO, Union
+
+import pytest
+
+from ert._c_wrappers.enkf import ErtConfig
+from ert.parsing import ConfigValidationError
+
+test_config_file_base = "test"
+test_config_filename = f"{test_config_file_base}.ert"
+
+
+@dataclass
+class FileDetail:
+    contents: str
+    is_executable: bool = False
+
+
+def write_files(
+    files: Union[dict, FileDetail], write_after: Callable[[TextIO], None] = None
+):
+    if files is not None:
+        for other_filename, content in files.items():
+            if isinstance(content, str):
+                with open(other_filename, mode="w", encoding="utf-8") as fh:
+                    fh.writelines(content)
+
+                    if write_after:
+                        write_after(fh)
+            else:
+                content: FileDetail
+                with open(other_filename, mode="w", encoding="utf-8") as fh:
+                    fh.writelines(content.contents)
+
+                if not content.is_executable:
+                    os.chmod(other_filename, stat.S_IREAD)
+
+
+def assert_that_config_leads_to_error(
+    config_file_contents: str,
+    expected_line: Optional[int],
+    expected_column: Optional[int],
+    expected_end_column: Optional[int],
+    expected_filename: str = "test.ert",
+    filename: str = "test.ert",
+    other_files: Union[dict, FileDetail] = None,
+    match: str = None,
+    write_after: Callable[[TextIO], None] = None,
+):
+    write_files(files={filename: config_file_contents}, write_after=write_after)
```

Comment:
This call can just be `pathlib.Path(filename).write_text(config_file_contents)`

Reply:
Done